### PR TITLE
Improved error message when a local resource of Devfile is not supported

### PR DESCRIPTION
**Pull request by sleshchenko**
### What does this PR do?
Improves error message when a local resource of Devfile is not supported
Previously we had the following message during factory accepting if Devfile is specified with content URL and needs local file content.

Previously
`http://che-multik.192.168.99.100.nip.io/f?url=https://raw.githubusercontent.com/eclipse/che/master/.devfile`
screenshot_20190222_134326

With these changes
`http://che-multik.192.168.99.100.nip.io/f?url=https://raw.githubusercontent.com/eclipse/che/master/.devfile`
screenshot_20190228_121234

Devfile API error message
```
Fetching content of file `.che-dev.yaml` specified in `local` field of tool `che-dev` 
is not supported. Please provide its content in `localContent` field. 
Cause: Devfile Service does not support fetching local file referenced 
in Devfile.
```

### What issues does this PR fix or reference?
N/A

#### Release Notes
N/A

#### Docs PR
N/A

### CODE REVIEW

**Review by skabashnyuk — APPROVED** [review]

**Comment by sleshchenko** [comment]
ci-test

**Review by garagatyi — APPROVED** [review]

**Review by mshaposhnik — APPROVED** [review]

**Comment by sleshchenko** [comment]
@eclipse/eclipse-che-qa Could you please review the latest test report https://github.com/eclipse/che/pull/12743#issuecomment-466455817?

**Comment by artaleks9** [comment]
ci-test

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevfileService.java [hunk 1]
```diff
@@ -50,6 +52,9 @@
 @Path("/devfile")
 public class DevfileService extends Service {
 
+  private static final String LOCAL_FILES_REFERENCES_ARE_NOT_SUPPORTED =
+      "Devfile Service does not support tools which require local files content. You can use factory from Github URL instead.";
```
Comment (metlos) [inline wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevfileService.java]: Are we limiting ourselves to github here on purpose?

File: wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/DefaultFactoryParameterResolver.java [hunk 2]
```diff
@@ -29,6 +30,10 @@
 @Singleton
 public class DefaultFactoryParameterResolver implements FactoryParametersResolver {
 
+  public static final String LOCAL_FILES_REFERENCES_ARE_NOT_SUPPORTED =
+      "Devfile requires local file content but this functionality is not supported when URL to raw content is specified. "
+          + "You can specify URL to Github repository instead.";
```
Comment (metlos) [inline wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/DefaultFactoryParameterResolver.java:35]: Is github going to be the only repo we're going to ever support?

Comment (sleshchenko) [reply wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/DefaultFactoryParameterResolver.java:35]: Currently, only Github is supported. Later any git hosting service may be implemented but it should provide access to files content without cloning it (like https://raw.githubusercontent.com).

File: wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevfileService.java (hunk 1 above)
Comment (sleshchenko) [reply wsmaster/che-core-api-devfile/src/main/java/org/eclipse/che/api/devfile/server/DevfileService.java]: Now, it's the only one git hosting service that we support for factories.

**Comment by artaleks9** [comment]
Selenium tests execution on Eclipse Che Multiuser on OCP (https://ci.codenvycorp.com/job/che-pullrequests-test-ocp/1585/) doesn't show any regression against this Pull Request.

**Comment by sleshchenko** [comment]
@metlos Could you please review again. Actual error messages are in PR description.

**Comment by sleshchenko** [comment]
@metlos I've merged PR since it contains small refactoring and it may influence other PRs. Even if new messages are not ideal but they are better then old ones.
Feel free to propose better messages and I'll create a new PR or contribute them on your own.

**Pull request merged**